### PR TITLE
Adding RSpec and initial tests

### PR DESCRIPTION
**Pull request by JackDanger**
This PR adds some basic tests to avoid regressions. It doesn't provide very wide test coverage but it should provide a place for any new contributors to add tests for their features to ensure we have a way to prevent regressions.

### CODE REVIEW

**Comment by JackDanger** [comment]
I've also configured Travis-ci.org to start running this project again.

File: ci [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -exuo pipefail
+
+bundle exec rspec
+#bundle exec rubocop -d
```
Comment (xaviershay) [inline ci]: why the comment? intentional?

**Comment by xaviershay** [comment]
LGTM

File: ci (hunk 1 above)
Comment (JackDanger) [inline ci]: Unfortunately `rubocop` isn't available on master yet so this doesn't run. The comment was an unfortunate median between adding Rubocop or skipping it for this PR (I've now removed the comment, will follow up).

**Comment by JackDanger** [comment]
Merging due to @xaviershay's LGTM

**Pull request merged**